packages/expo-image-manipulator/android/src/main/java/expo/modules/imagemanipulator/ImageManipulatorModule.kt

```diff
@@ -3,88 +3,73 @@ package expo.modules.imagemanipulator
-import expo.modules.core.ExportedModule
-import expo.modules.core.ModuleRegistry
-import expo.modules.core.ModuleRegistryDelegate
-import expo.modules.core.Promise
-import expo.modules.core.arguments.ReadableArguments
-import expo.modules.core.interfaces.ExpoMethod
+import expo.modules.kotlin.Promise
+import expo.modules.kotlin.exception.Exceptions
+import expo.modules.kotlin.modules.Module
+import expo.modules.kotlin.modules.ModuleDefinition
 import java.io.ByteArrayOutputStream
 import java.io.File
 import java.io.FileOutputStream
-import java.util.*
 
-private const val TAG = "ExpoImageManipulator"
 private const val ERROR_TAG = "E_IMAGE_MANIPULATOR"
 
-class ImageManipulatorModule(
-  context: Context,
-  private val moduleRegistryDelegate: ModuleRegistryDelegate = ModuleRegistryDelegate()
-) : ExportedModule(context) {
-  private val mImageLoader: ImageLoaderInterface by moduleRegistry()
+class ImageManipulatorModule : Module() {
+  private val context: Context
+    get() = appContext.reactContext ?: throw Exceptions.ReactContextLost()
 
-  private inline fun <reified T> moduleRegistry() = moduleRegistryDelegate.getFromModuleRegistry<T>()
-  override fun getName() = TAG
+  override fun definition() = ModuleDefinition {
+    Name("ExpoImageManipulator")
 
-  override fun onCreate(moduleRegistry: ModuleRegistry) {
-    moduleRegistryDelegate.onCreate(moduleRegistry)
-  }
-
-  @ExpoMethod
-  fun manipulateAsync(uri: String, actionsRaw: ArrayList<Any?>, saveOptionsRaw: ReadableArguments, promise: Promise) {
-    val saveOptions = SaveOptions.fromArguments(saveOptionsRaw)
-    val actions = Actions.fromArgument(actionsRaw)
-    mImageLoader.loadImageForManipulationFromURL(
-      uri,
-      object : ResultListener {
-        override fun onSuccess(bitmap: Bitmap) {
-          runActions(bitmap, actions, saveOptions, promise)
-        }
+    AsyncFunction("manipulateAsync") { uri: String, actionsRaw: List<ManipulateAction>, saveOptions: SaveOptions, promise: Promise ->
+      val actions = Actions.fromArgument(actionsRaw)
+      appContext.imageLoader?.loadImageForManipulationFromURL(
+        uri,
+        object : ResultListener {
+          override fun onSuccess(bitmap: Bitmap) {
+            runActions(bitmap, actions, saveOptions, promise)
+          }
 
-        override fun onFailure(cause: Throwable?) {
-          // No cleanup required here.
-          val basicMessage = "Could not get decoded bitmap of $uri"
-          if (cause != null) {
-            promise.reject("${ERROR_TAG}_DECODE", "$basicMessage: $cause", cause)
-          } else {
-            promise.reject("${ERROR_TAG}_DECODE", "$basicMessage.")
+          override fun onFailure(cause: Throwable?) {
+            // No cleanup required here.
+            val basicMessage = "Could not get decoded bitmap of $uri"
+            if (cause != null) {
+              promise.reject("${ERROR_TAG}_DECODE", "$basicMessage: $cause", cause)
```

Comment:
You can use the `ImageDecodeException` here too. Just pipe the exception to the `CodedException` using the contractor. You did similar thing in the `NetworkModule` ;)